### PR TITLE
Using pedantic

### PR DESCRIPTION
**Pull request by jtlap**
It's the first step of the decorators reorganization

### CODE REVIEW

File: include/boost/simd/arch/common/scalar/function/arg.hpp [hunk 1]
```diff
@@ -18,34 +18,30 @@
...
+                             , bs::pedantic_tag
                              , bd::scalar_< bd::floating_<A0> >
                              )
      {
-       BOOST_FORCEINLINE A0 operator() ( A0 a0) const BOOST_NOEXCEPT
+       BOOST_FORCEINLINE A0 operator() (const pedantic_tag &,
+                                        A0 a0) const BOOST_NOEXCEPT
        {
-         A0 r = if_else_zero(is_ltz(a0),Pi<A0>());
+         std::cout << "a0 " << a0 << std::endl;
```
Comment (jfalcou) [inline include/boost/simd/arch/common/scalar/function/arg.hpp]: nope ;)

File: include/boost/simd/arithmetic.hpp [hunk 2]
```diff
@@ -141,14 +141,14 @@ namespace boost { namespace simd
...
+      code you can use pedantic_(fma) (although it can be expansive) or
       (generally still more expansive) std_(fma) by using the decorators.
 
-      @ref two_add, @ref two_prod and @ref two_split are used internally in @ref conformant_(fma) and
+      @ref two_add, @ref two_prod and @ref two_split are used internally in @ref pedantic_(fma) and
       can be useful in searching extra-accuracy in other circumstances as double-double
       computations.
 
-      @ref conformant_(fma) decis never used internally by Boost.SIMD
+      @ref pedantic_(fma) decis never used internally by Boost.SIMD
```
Comment (jfalcou) [inline include/boost/simd/arithmetic.hpp]: "decis" ??? "is" you mean ?

File: include/boost/simd/function/atan2.hpp [hunk 3]
```diff
@@ -62,7 +62,7 @@ namespace boost { namespace simd
 
     std_ for floating entries provides access to std::atan2
 
-    fast_ is speedier but as indicated is not fully conformant to the standard
+    fast_ is speedier but as indicated is not fully pedantic to the standard
```
Comment (jfalcou) [inline include/boost/simd/function/atan2.hpp]: wrong replace

File: include/boost/simd/function/fma.hpp [hunk 4]
```diff
@@ -45,15 +45,15 @@ namespace boost { namespace simd
     in terms of performance for floating ones (i.e. if the system has the hard
     wired capability).
 
-    If you need conformant fma capabilities in all circumstances in your own
-    code you can use the conformant_ or  std_ decorator
+    If you need pedantic fma capabilities in all circumstances in your own
+    code you can use the pedantic_ or  std_ decorator
     (although both can can be very expansive).
```
Comment (jfalcou) [inline include/boost/simd/function/fma.hpp]: "expensive"

File: include/boost/simd/arch/common/scalar/function/arg.hpp (hunk 1 above)
Comment (jtlap) [reply include/boost/simd/arch/common/scalar/function/arg.hpp]: argh

File: include/boost/simd/function/fma.hpp (hunk 4 above)
Comment (jtlap) [reply include/boost/simd/function/fma.hpp]: done

File: include/boost/simd/arithmetic.hpp (hunk 2 above)
Comment (jtlap) [reply include/boost/simd/arithmetic.hpp]: yes done

File: include/boost/simd/function/atan2.hpp (hunk 3 above)
Comment (jtlap) [reply include/boost/simd/function/atan2.hpp]: corrected

**Pull request merged**